session/pingpong/consumer_balance_tracker.go

```diff
@@ -144,7 +147,32 @@ func (cbt *ConsumerBalanceTracker) handleTopUpEvent(id string) {
 func (cbt *ConsumerBalanceTracker) handleRegistrationEvent(event registry.RegistrationEventPayload) {
 	switch event.Status {
 	case registry.RegisteredConsumer, registry.RegisteredProvider:
-		cbt.updateBalanceFromAccountant(event.ID)
+		var boff backoff.BackOff
+		eback := backoff.NewExponentialBackOff()
+		eback.MaxElapsedTime = time.Minute
+		eback.InitialInterval = time.Second * 2
+
+		boff = backoff.WithMaxRetries(eback, 10)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		go func() {
+			defer cancel()
```

Comment:
Why is this second defer cancel needed and in separate goroutine? Maybe add some comment as it is a bit hard to understand.